upf/upf_pfcp_server.c

```diff
@@ -414,8 +411,14 @@ request_t1_expired (u32 seq_no)
 	     format_node_id, &n->node_id, format_ip46_address, &n->lcl_addr,
 	     IP46_TYPE_ANY, format_ip46_address, &n->rmt_addr, IP46_TYPE_ANY);
 
+	  /* This will free the message at the end of pfcp_process() loop */
 	  pfcp_release_association (n);
 	}
+      else
+	{
+	  hash_unset (psm->request_q, msg->seq_no);
+	  pfcp_msg_pool_put (psm, msg);
```

Comment:
Instead of adding else clause and moving the msg release there, wouldn't it be simpler to leave it where it is and to copy the `msg->node` value to local variable and use that in the association setup?